zerver/lib/push_notifications.py

```diff
@@ -1103,6 +1104,17 @@ def handle_push_notification(user_profile_id: int, missed_message: Dict[str, Any
         user_group = UserGroup.objects.get(id=mentioned_user_group_id, realm=user_profile.realm)
         mentioned_user_group_name = user_group.name
```

Comment:
Should this logic be giving precedence to personal mentions?  Oh, maybe it does already, in  `get_user_group_mentions_data`. This is potentially confusing, and we could make it less so if we move this function:
```
def get_mentioned_user_group_name(                                                                       
    messages: List[Dict[str, Any]], user_profile: UserProfile                                            
) -> Optional[str]:                                                                                      
    """Returns the user group name to display in the email notification                                  
    if user group(s) are mentioned.                                                                      
                                                                                                         
    This implements the same algorithm as get_user_group_mentions_data                                   
    in zerver/lib/notification_data.py, but we're passed a list of                                       
    messages instead.                                                                                    
    """                                                                                                  
```

to `zerver/lib/notification_data.py` so that they're in the same place; maybe that's a useful commit to add this PR?

Reply:
Yes, it does. The logic is just not straightforward. As this property of `mentioned_user_group_id` is implicitly derived from the map that `get_user_group_mentions_data` returns, which is passed from the `SendMessageRequest` to a event dict. Probably also some comments on that matter help too.

Reply:
Yeah, I think comments are a good idea here.